irods/helpers/__init__.py

```diff
@@ -0,0 +1,50 @@
+import functools
+import inspect
+import types
+import contextlib
+from ..test.helpers import make_session as make_test_session
+from irods.message import (ET, XML_Parser_Type)
+
+
+# See https://stackoverflow.com/questions/67028108/overriding-function-signature-in-help-when-using-functools-wraps
+def update_func(f, func_name='', update_kwargs: dict = None):
+    """Based on http://stackoverflow.com/a/6528148/190597 (Glenn Maynard)"""
+    g = types.FunctionType(
+            code=f.__code__,
+            globals=f.__globals__.copy(),
+            name=f.__name__,
+            argdefs=f.__defaults__,
+            closure=f.__closure__
+    )
+
+    g = functools.update_wrapper(g, f)
+    g.__signature__ = inspect.signature(g)
+    g.__kwdefaults__ = (f.__kwdefaults__ or {}).copy()
+
+    # Adjust your arguments
+    for key, value in (update_kwargs or {}).items():
+        g.__kwdefaults__[key] = value
+        g.__signature__.parameters[key]._default = value
+
+    g.__name__ = func_name or g.__name__
+    return g
```

Comment:
Tell me about this function.

Is it meant for public use?

Reply:
On inspection it isn't dangerous, and it is the simplest way to arrive at a help text for `irods.helpers.make_session` where the default value of the `test_server_version` parameter reflects the truth on the ground...

I didn't really want to use it, but seemingly had to.

Reply:
I could add a test verifying the function of this one.
Or, we could do some kind of funky regex-tricks to play with the __doc__ text, a little bit more brittle but clearer in intent:
```
def make_session( ...) : return make_test_session( ...)
make_session.__doc__ = re.sub('(test_server_version\s*=)\s*\w+',r'\1 False',make_test_session.__doc__)
```

Reply:
Unless `update_func` is meant for public use, I don't think a test is necessary.

Is it for public use?

Reply:
If it is for public use, that's only by virtue of being  exposed on stack exchange and with caveats.  It's labelled hacky.
One alternative is (as he states in the link i've put in the source code there)  **(A)** just refrain from caring if the help text is actually accurate or not, and do nothing.  Middle ground would be: **(B)** just twiddle the help text with a regex like I'm now doing for Python2.
I could **(C)** leave everything as it is, and if  confidence is low because of the various factors involved, write a perfunctory test.  (So far it seems to work fine, from manually testing.) 

I defer in what path should be taken, being agnostic on this point.  I don't know how much we care whether documentation is accurate in this case.  The function works as is, or seems to, even having been processed in such a "hackish" way via the `update_func`

Reply:
I admit this solution (C) is quite ugly.

Reply:
Or... (D) the help text could be a simple one that refers users to the irods.test.helpers version, noting that it should be consulted for parameter descriptions but that the default value for test_server_version is different.

(E), I suppose, would be just to copy the help text verbatim from one place to the other in its entirety, then make the tiny tweak of adjusting for the different default value.  But that's two sources of truth.

I think I prefer  (B), actually.  The only thing it really lacks is the assuredness of whether the __doc__ attribute will forever be a mutable one.

Reply:
@korydraughn sorry for the spam, but I'm probably not well versed on what the significance or meaning of "for public use" actually is. : )

Reply:
And finally, in favor of (B), modifying __doc__ directly and with the regex tweak, here's this:
https://stackoverflow.com/questions/66586795/can-you-define-doc-directly

Reply:
I was referring to whether `update_func` should have a leading underscore so users of the PRC know whether they are allowed to use it or not.

So in this case, "public" means users of the PRC are allowed to import/use `update_func`. "private" means the function is renamed to `_update_func` indicating it's an implementation detail and users of the PRC must NOT use it.

Reply:
Ah, ok... I see.  I think no, it would be better to keep it a private function.

Reply:
In the end, decided to get rid of `update_func` as it wasn't having the intended effect. (That is to alter both the function's default argument, as well as to have that reflected in the docstring.)  So, resolving this.